### ISSUE TITLE
Sort-Object should use the IStructuralComparable interface, if implemented

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#18389]
### Summary of the new feature / enhancement

Currently, the `System.IComparable` interface is honored when `Sort-Object` compares two input objects.

It would be beneficial to also honor the `System.Collections.IStructuralComparable` interface for collection-like types that implement it, notably `[array]`s.

That is, the following should work as expected:

```powershell
# SHOULD be '[10 42]', '[9 42]', but currently isn't, because the input arrays are effectively
# *stringified* and then compared.
(9, 42), (10, 42) |  Sort-Object -Descending | ForEach-Object { "[$_]" }
```

Note that there are collection-like types that implement _both_ interfaces, in which case sorting  does work, notably `System.ValueTuple`:

```powershell
# OK ->  '[10 42]', '[9 42]'
(9, 42), (10, 42) |  Sort-Object -Descending { [ValueTuple]::Create.Invoke($_) }  | ForEach-Object { "[$_]" }
```

### Proposed technical implementation details (optional)

Note that, more generally, the change to support `IStructuralComparable` should be made to whatever engine logic is at the heart of trivalent object comparisons, irrespective of a specific context such as `Sort-Object`  - perhaps [`PSObject.CompareTo()`](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.psobject.compareto)?

**Comment by iRon7** [comment]
Coming from: [sorting arrays of arrays of ints (2-d array) does not work (it sorts as strings; 1-d arrays work)](https://stackoverflow.com/questions/79863749/sorting-arrays-of-arrays-of-ints-2-d-array-does-not-work-it-sorts-as-strings).
I wasn't aware of this enhancement request, but taken the fact that PowerShell is a loosely language, I suspect that this isn't really feasible.
Take the following list of (`2`) arrays as an example:

```PowerShell
                  # or even:
$a = 1, 2         # [int[]](1, 2)
$b = 'a', 'b'     # [string[]]('a', 'b')
$a, $b | Sort-Object -Descending | ForEach-Object { "[$_]" }
```

How would you compare the items: `$a.CompareTo($b)` or `$b.CompareTo($a)`? Also knowing that during the sorting algorithm process of a longer list, the order of `$a` and `$b` might swap continuously.
In other words, I think that it is required to explicitly define the comparer as requested in the ([rejected](https://github.com/PowerShell/PowerShell/issues/24560#issuecomment-2637580633)) propose #24560, where this one is (@SteveL-MSFT) <kbd>[Up-for-Grabs](https://github.com/PowerShell/PowerShell/issues?q=state%3Aopen%20label%3A%22Up-for-Grabs%22)</kbd>.
